File: backend/app/utils/calculation.py
def completion_rate(completed: int, total: int) -> float:
    if total <= 0:
        return 0.0
    return round((completed / total) * 100, 2)


def mean(values: list[float]) -> float:
    if not values:
        return 0.0
    return sum(values) / len(values)

File: frontend/src/utils/calculation.ts
export const calculateCompletionRate = (completed: number, total: number): number => {
  if (total <= 0) {
    return 0;
  }

  return (completed / total) * 100;
};

export const average = (values: number[]): number => {
  if (!values.length) {
    return 0;
  }

  return values.reduce((sum, current) => sum + current, 0) / values.length;
};
